spynnaker/pyNN/models/neuron/master_pop_table.py

```diff
@@ -315,57 +316,59 @@ class MasterPopTableAsBinarySearch(object):
         "__entries",
         "__n_addresses"]
 
-    MAX_ROW_LENGTH_ERROR_MSG = (
-        "Only rows of up to {} entries are allowed".format(
-            POP_TABLE_MAX_ROW_LENGTH))
-
-    OUT_OF_RANGE_ERROR_MESSAGE = (
-        "Address {} is out of range for this population table!")
-
-    # Over-scale of estimate for safety
-    UPPER_BOUND_FUDGE = 2
-
     def __init__(self):
         self.__entries = None
         self.__n_addresses = 0
 
-    def get_master_population_table_size(self, in_edges):
+    @staticmethod
+    def get_master_population_table_size(incoming_projections):
         """ Get the size of the master population table in SDRAM.
 
-        :param iterable(~pacman.model.graphs.application.ApplicationEdge) \
-                in_edges:
-            The edges arriving at the vertex that are to be handled by this
-            table
+        :param iterable(PyNNProjectionCommon) incoming_projections:
+            The projections arriving at the vertex that are to be handled by
+            this table
         :return: the size the master pop table will take in SDRAM (in bytes)
         :rtype: int
         """
+        # There will be an address list entry for each incoming projection
+        n_entries = len(incoming_projections)
 
-        # Entry for each edge - but don't know the edges yet, so
-        # assume multiple entries for each edge
+        # Count the pre-machine-vertices
         n_vertices = 0
-        n_entries = 0
-        for in_edge in in_edges:
-            if isinstance(in_edge, ProjectionApplicationEdge):
-                slices, is_exact = (
-                    in_edge.pre_vertex.splitter.get_out_going_slices())
-                if is_exact:
-                    n_vertices += len(slices)
-                    n_entries += len(in_edge.synapse_information)
+        seen_edges = set()
+        for proj in incoming_projections:
+            in_edge = proj._projection_edge
+
+            # If we haven't seen this edge before, add it in
+            if in_edge not in seen_edges:
+                seen_edges.add(in_edge)
+                vertex = in_edge.pre_vertex
+
+                # If the pre-vertex not been partitioned, estimate
+                if not vertex.machine_vertices:
```

Comment:
that is not a great check for not being partitioned. and givne the vertex.splitter.get_out_going_slices() returns 2 elements. 1 the slices, but 2 a BOOL THAT SAYS ITS AN ESTIMATE OR NOT! 
so line 349 is now a length of a tuple. but also, means you could use that bool. or heck, you could just use the line 348 for either.